File: Embodied-AI/ROS/catkin_ws/src/map_pkg/src/map_pub_node.cpp
#include <ros/ros.h>
#include <nav_msgs/OccupancyGrid.h>

int main(int argc, char  *argv[])
{
    ros::init(argc, argv, "map_pub_node");
    ros::NodeHandle nh;
    ros::Publisher map_pub = nh.advertise<nav_msgs::OccupancyGrid>("/map", 1);
    ros::Rate loop_rate(10);
    while(ros::ok())
    {
        nav_msgs::OccupancyGrid msg;
        msg.header.frame_id = "map";
        msg.header.stamp = ros::Time::now();
        //地图相对坐标系原点的偏移量
        msg.info.origin.position.x = 1.0; 
        msg.info.origin.position.y = 0.0;
        msg.info.resolution = 1.0; //栅格的长度
        msg.info.width = 4;
        msg.info.height = 2;

        msg.data.resize(msg.info.width * msg.info.height);
        msg.data[0]=90;
        msg.data[1]=100;
        msg.data[2]=0;
        msg.data[3]=-1;
        map_pub.publish(msg);
        loop_rate.sleep();

    }
    return 0;
}

File: Embodied-AI/ROS/catkin_ws/src/map_pkg/scripts/map_pub_node.py
#! /usr/bin/env python3
# coding: utf-8
import rospy
from nav_msgs.msg import OccupancyGrid

if __name__ == '__main__':
    rospy.init_node('map_pub_node')
    pub = rospy.Publisher('/map', OccupancyGrid, queue_size=10)
    rate = rospy.Rate(10)
    while not rospy.is_shutdown():
        msg = OccupancyGrid()
        msg.header.stamp = rospy.Time.now()
        msg.header.frame_id = 'map'

        msg.info.resolution = 1
        msg.info.width = 4
        msg.info.height = 2
        msg.info.origin.position.x = 0
        msg.info.origin.position.y = 0

        msg.data = [0]*4*2
        msg.data[0]=100
        msg.data[1]=90
        msg.data[2]=0
        msg.data[3]=-1

        pub.publish(msg)
        rate.sleep()
        